pkgs/development/interpreters/python/buildpypi.nix

```diff
@@ -0,0 +1,35 @@
+/* This function wraps buildPythonPackage and uses pre-fetched data from PyPI to build a package.
+*/
+{ buildPythonPackage
+, fetchurl
+, fetchgit
+, lib
+}:
+
+
+{ name
+, version ? ""
+, ... } @ attrs:
+
+let
+  pypi-src = fetchgit {
```

Comment:
> Is fetchgit required here?

Since GitHub can produce archives for a specific revision we could use `fetchurl`. But, where we really want to go to is fetchTarbal` along with a sha256. That way, Hydra can evaluate it. Without fetchTarball + sha256 Hydra cannot evaluate it. That's the reason why this is on hold. See also [my latest mail](https://mail-archive.com/nix-dev@lists.science.uu.nl/msg27110.html).

Another option to reduce the size is to work with two repositories, one that has a JSON for each package, and another one that has package names along with hashes of the JSON files. This repo we would then refer to from Nixpkgs. It would replace downloading a big file with (possibly many) tiny ones.